api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -539,4 +540,25 @@ public Profile updateAccountProperties(AccountPropertyUpdate request) {
 
     return getProfile(dbUser);
   }
+
+  public void validateTermsOfService(Integer tosVersion) {
+    if (tosVersion == null) {
+      throw new BadRequestException("Terms of Service version is NULL");
+    }
+    if (tosVersion != CURRENT_TERMS_OF_SERVICE_VERSION) {
+      throw new BadRequestException("Terms of Service version is not up to date");
+    }
+  }
+
+  public void validateTermsOfService(DbUser dbUser) {
```

Comment:
moved to ProfileService partially so it would have access to userTermsOfServiceDao

Reply:
Should we add a comment saying this method is for existing users

Reply:
I think it's a little different than that.  It's for users who have been partially created (Google, DB) but not fully (Terra)

I suspect we'll need to change this a bit for fully-created existing users in the next story